NetworkPkg/NvmeOfDxe/NvmeOfMisc.c

```diff
@@ -1824,3 +1914,127 @@ NvmeOfBeforeEBS (
   //
   gDriverInRuntime = TRUE;
 }
+
+/**
+  Create and initialize the NVMe-oF Global data with default values.
+
+  If the NV variable 'NvmeofGlobalData' is not found, data will be created and
+  default parameters assigned.
+  If the NV data is found it will be kept to ensure persistence of user-configured
+  data (HostNQN, HostID). Default parameters will still be set to maintain
+  consistency with the driver or apply updated values (such as driver version).
+
+  @retval EFI_SUCCESS             The Global data has been created successfully.
+  @retval EFI_UNSUPPORTED         Located existing data with unexpected size.
+  @retval Others                  Failed to create Global data.
+
+**/
+EFI_STATUS
+NvmeOfInitializeGlobalNvData (
+  VOID
+  )
+{
+  NVMEOF_GLOBAL_DATA  *NvmeOfGlobalData;
+  EFI_GUID            *SystemGuid;
+  EFI_STATUS          Status = EFI_SUCCESS;
+  UINTN               Size;
+
+  SystemGuid = NULL;
+
+  NvmeOfGlobalData = NvmeOfGetVariableAndSize (
+                       L"NvmeofGlobalData",
+                       &gNvmeOfConfigGuid,
+                       &Size
+                       );
+  if ((Size != sizeof (NVMEOF_GLOBAL_DATA)) && (Size != 0)) {
+    DEBUG ((
+      DEBUG_ERROR,
+      "%a: Located variable (NvmeOfGlobalData) with Guid (%d) "
+      "has unexpected data size.\n",
+      __FUNCTION__,
+      &gNvmeOfConfigGuid
+      ));
+    return EFI_UNSUPPORTED;
+  }
+
+  if ((NvmeOfGlobalData == NULL) || (Size == 0)) {
+    NvmeOfGlobalData = AllocateZeroPool (sizeof (NVMEOF_GLOBAL_DATA));
+    if (NvmeOfGlobalData == NULL) {
+      return EFI_OUT_OF_RESOURCES;
+    }
+  }
+
+  //
+  // Set Host ID and/or Host NID according to system UUID if not already set
+  //
+  if (  (NvmeOfGlobalData->NvmeofHostId[0] == '\0')
+     || (NvmeOfGlobalData->NvmeofHostNqn[0] == '\0'))
+  {
+    SystemGuid = AllocateZeroPool (sizeof (EFI_GUID));
+    if (SystemGuid == NULL) {
+      return EFI_OUT_OF_RESOURCES;
+    }
+
+    Status = NetLibGetSystemGuid (SystemGuid);
+    if (EFI_ERROR (Status)) {
+      DEBUG ((
+        DEBUG_ERROR,
+        "%a: Cannot read system GUID.\n",
+        __FUNCTION__,
+        &gNvmeOfConfigGuid
```

Comment:
Only one parameter in this debug call, is that coding error?

Reply:
Thanks! Resolved in [fa5d8f5](https://github.com/timberland-sig/edk2/pull/20/commits/fa5d8f541602b120fb5fd9c91a31097589874740)